server/test/repository/ApplicationEventRepositoryTest.java

```diff
@@ -149,4 +149,63 @@ public void getEvents() {
     // NoteEvent wasn't set and should be available as an empty Optional.
     assertThat(gotEvent.getDetails().noteEvent()).isNotPresent();
   }
+
+  @Test
+  public void test_setNoteEvent() {
+    // Setup
+    Instant startInstant = Instant.now();
+    ProgramModel program = resourceCreator.insertActiveProgram("Program");
+    AccountModel actor = resourceCreator.insertAccount();
+    ApplicantModel applicant = resourceCreator.insertApplicantWithAccount();
+    ApplicationModel application = resourceCreator.insertActiveApplication(applicant, program);
+
+    repo.setNote(application, ApplicationEventDetails.NoteEvent.create("some note"), actor);
+
+    // Execute
+    ImmutableList<ApplicationEventModel> applicationEvents =
+        repo.getEventsOrderByCreateTimeDesc(application.id);
+
+    // Verify
+    assertThat(applicationEvents).hasSize(1);
+    ApplicationEventModel gotEvent = applicationEvents.get(0);
```

Comment:
**nitpick:** Maybe `gotEvent` here and below can be called `firstAppEvent` or something a bit clearer?